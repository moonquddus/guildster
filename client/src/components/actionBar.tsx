import styled from 'styled-components';

export default styled.div`
  width: 100%;
  height: 80px;
  display: flex;
  align-items: center;
  position: absolute;
  bottom: 0;
  left: 0;
  background: #3d414c;
  vertical-align: middle;
  padding: 0px 30px;
`;
