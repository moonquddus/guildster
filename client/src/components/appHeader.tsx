import styled from 'styled-components';

export const SubHeader = styled.h2`
font-size: 20px;
margin: 0;
text-align: center;
text-transform: uppercase;
letter-spacing: -1px;
`;

export default styled.h1`
  font-size: 30px;
  margin: 0 0 60px;
  text-align: center;
  text-transform: uppercase;
  letter-spacing: -1px;
`;
