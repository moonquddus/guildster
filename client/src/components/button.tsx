import styled from 'styled-components';

// TODO: Use ThemeProvider rather than hacking some variables together
/*
Const buttonThemes = {
  redTheme: {
    background: 'linear-gradient(0deg, rgba(174,22,0,1) 0%, rgba(174,22,0,1) 50%, rgba(203,44,22,1) 50%, rgba(203,44,22,1) 100%)',
    border: 'linear-gradient(0deg, rgba(199,40,18,1) 0%, rgba(199,40,18,1) 50%, rgba(224,65,43,1) 50%, rgba(224,65,43,1) 100%)',
  },
  long: {
    minWidth: '200px',
  },
  fullWidth: {
    minWidth: '100%',
  },
};
*/

export default styled.button`
  justify-self: ${(props) => props.theme.align || 'center'};
  position: relative;
  background: linear-gradient(0deg, rgba(174,22,0,1) 0%, rgba(174,22,0,1) 50%, rgba(203,44,22,1) 50%, rgba(203,44,22,1) 100%);
  border: solid 6px;
  border-image: linear-gradient(0deg, rgba(199,40,18,1) 0%, rgba(199,40,18,1) 50%, rgba(224,65,43,1) 50%, rgba(224,65,43,1) 100%);
  border-image-slice: 1;
  color: #ffdfb0;
  text-transform: uppercase;
  font-size: 14px;
  letter-spacing: -1px;
  outline: none;
  box-shadow: 3px 3px 0px 0px rgb(0, 0, 0, 0.2);
  margin: auto;
  padding: ${(props) => props.theme.height || '10px'} 15px;
  font-weight: bold;
  font-size: 20px;
  cursor: pointer;
  text-align: center;
  min-width: auto;
  width: ${(props) => props.theme.width || 'auto'};

  &:hover {
    color: #f7d6a6;
  }

  &:active {
    transform: translateY(2px);
  }
`;
