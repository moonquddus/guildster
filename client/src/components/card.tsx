import styled from 'styled-components';

export default styled.main`
  width: 90%;
  max-width: ${(props) => props.theme.width || '1000px'};
  min-height: 50vh;
  margin: 10% auto 0;
  background: #efc78c;
  box-shadow: 0px 0px 10px 0px;
  padding: 60px;
  position: relative;
`;
