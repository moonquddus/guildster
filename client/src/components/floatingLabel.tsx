import styled from 'styled-components'

export default styled.div`
  margin-top: 10px;
  position: relative;

  & label {
    position: absolute;
    top: 10px;
    left: 15px;
    opacity: 0;
    transition: all .3s ease;
    font-size: 16px;
    z-index: -1;
  }

  & input {
    transition: 0.3s linear;

    &:not(:placeholder-shown) + label {
      z-index: 1;
      font-size: 12px;
      opacity: 0.8;
      transform: translate(0px, -25px)   
    }
  }
`;