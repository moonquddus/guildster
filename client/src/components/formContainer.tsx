import styled from 'styled-components'

export default styled.div`
  margin: 60px auto 60px;
  max-width: 400px;
`;