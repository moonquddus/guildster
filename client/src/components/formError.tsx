import styled from 'styled-components'

export default styled.span`
  font-size: 12px;
  color: red;
  opacity: 0.8;
  width: 100%;
  position: absolute;
  top: 40px;
  left: 0;
  margin: 0 15px;
`;