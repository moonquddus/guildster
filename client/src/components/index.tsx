import ActionBar from './actionBar'
import AppHeader, { SubHeader } from './appHeader'
import Button from './button'
import Card from './card'
import FloatingLabel from './floatingLabel'
import FormContainer from './formContainer'
import Input from './input'
import Label from './label'
import { SimpleGrid, Left, Right } from './simpleGrid'
import Select from './select'
import HealthBar from './healthBar'

export default { ActionBar, AppHeader, SubHeader, Button, Card, FloatingLabel, FormContainer, Input, Label, SimpleGrid, Left, Right, Select, HealthBar }