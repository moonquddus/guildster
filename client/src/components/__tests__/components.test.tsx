import React from 'react'
import renderer from 'react-test-renderer'

import components from '../'

const { HealthBar, StatBar, FormError } = components

test('ActionBar renders correctly', () => {
  const { ActionBar } = components
  const output = renderer
    .create(<ActionBar>Testing</ActionBar>)
    .toJSON();
  expect(output).toMatchSnapshot()
});

test('AppHeader renders correctly', () => {
  const { AppHeader } = components
  const output = renderer
    .create(<AppHeader>My Header</AppHeader>)
    .toJSON();
  expect(output).toMatchSnapshot()
})

test('SubHeader renders correctly', () => {
  const { SubHeader } = components
  const output = renderer
    .create(<SubHeader>My Subheader</SubHeader>)
    .toJSON();
  expect(output).toMatchSnapshot()
})

test('Button renders correctly', () => {
  const { Button } = components
  const output = renderer
    .create(<Button type='button'>My Button</Button>)
    .toJSON();
  expect(output).toMatchSnapshot()
})

test('Card renders correctly', () => {
  const { Card } = components
  const output = renderer
    .create(<Card>This is a card.</Card>)
    .toJSON();
  expect(output).toMatchSnapshot()
})

test('Input w/ FloatingLabel renders correctly', () => {
  const { FormContainer, FloatingLabel, Input, Label } = components
  const output = renderer
    .create(
    <FormContainer>
      <FloatingLabel>
        <Input type='text' name='test' placeholder='Test Input' />
        <Label htmlFor='test'>Test Input:</Label>
      </FloatingLabel>
    </FormContainer>
    )
    .toJSON();
  expect(output).toMatchSnapshot()
})

test('Select renders correctly', () => {
  const { FormContainer, Select } = components
  const options = [
    {value: 'test1', name: 'Testing 1'},
    {value: 'test2', name: 'Testing 2'},
    {value: 'test3', name: 'Testing 3'}
  ]
  const setOption = () => false
  const output = renderer
    .create(
      <FormContainer>
        <Select label='Class' options={options} selectedOption='test2' setSelectedOption={setOption} />
      </FormContainer>
    )
    .toJSON();
  expect(output).toMatchSnapshot()
})

test('SimpleGrid renders correctly', () => {
  const { SimpleGrid, Left, Right } = components
  const output = renderer
    .create(
      <SimpleGrid>
        <Left>
          The left-hand side of the grid
        </Left>
        <Right>
          The right-hand side of the grid
        </Right>
      </SimpleGrid>
    )
    .toJSON();
  expect(output).toMatchSnapshot()
})

test('HealthBar renders at 100%', () => {
  const { HealthBar } = components
  const output = renderer
    .create(<HealthBar hp={100} max={100} />)
    .toJSON();
  expect(output).toMatchSnapshot()
})

test('HealthBar renders at 50%', () => {
  const { HealthBar } = components
  const output = renderer
    .create(<HealthBar hp={50} max={100} />)
    .toJSON();
  expect(output).toMatchSnapshot()
})

test('HealthBar renders at 0%', () => {
  const { HealthBar } = components
  const output = renderer
    .create(<HealthBar hp={0} max={100} />)
    .toJSON();
  expect(output).toMatchSnapshot()
})

test('StatBar renders at 100%', () => {
  const { StatBar } = components
  const output = renderer
    .create(<StatBar label='HIGH STAT' stat={100} max={100} />)
    .toJSON();
  expect(output).toMatchSnapshot()
})

test('StatBar renders at 50%', () => {
  const { StatBar } = components
  const output = renderer
    .create(<StatBar label='MID STAT' stat={50} max={100} />)
    .toJSON();
  expect(output).toMatchSnapshot()
})

test('StatBar renders at 0%', () => {
  const { StatBar } = components
  const output = renderer
    .create(<StatBar label='LOW STAT' stat={0} max={100} />)
    .toJSON();
  expect(output).toMatchSnapshot()
})
