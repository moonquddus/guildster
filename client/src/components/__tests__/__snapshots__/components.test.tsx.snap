// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar renders correctly 1`] = `
<div
  className="sc-bdnylx bYwksk"
>
  Testing
</div>
`;

exports[`AppHeader renders correctly 1`] = `
<h1
  className="sc-dlnjPT bqwPji"
>
  My Header
</h1>
`;

exports[`Button renders correctly 1`] = `
<button
  className="sc-hKFyIo idwWRH"
  type="button"
>
  My Button
</button>
`;

exports[`Card renders correctly 1`] = `
<main
  className="sc-eCApGN fPCvFY"
>
  This is a card.
</main>
`;

exports[`HealthBar renders at 0% 1`] = `
<span
  className="sc-ksluoS kAtFPB"
>
  <span
    className="sc-hBMVcZ bgyveJ"
  />
</span>
`;

exports[`HealthBar renders at 50% 1`] = `
<span
  className="sc-ksluoS kAtFPB"
>
  <span
    className="sc-hBMVcZ ineYPS"
  />
</span>
`;

exports[`HealthBar renders at 100% 1`] = `
<span
  className="sc-ksluoS kAtFPB"
>
  <span
    className="sc-hBMVcZ fKfxhH"
  />
</span>
`;

exports[`Input w/ FloatingLabel renders correctly 1`] = `
<div
  className="sc-gKAblj dIppkU"
>
  <div
    className="sc-jSFkmK cWyYbt"
  >
    <input
      className="sc-iCoHVE krsFLN"
      name="test"
      placeholder="Test Input"
      type="text"
    />
    <label
      className="sc-fujyUd dgolgJ"
      htmlFor="test"
    >
      Test Input:
    </label>
  </div>
</div>
`;

exports[`Select renders correctly 1`] = `
<div
  className="sc-gKAblj dIppkU"
>
  <div
    className="sc-iqAbSa tIQUW"
  >
    <label
      className="sc-fujyUd dgolgJ"
    >
      Class
    </label>
    <div
      className="sc-crzoUp ctHhyA"
      onClick={[Function]}
    >
      Testing 2
    </div>
  </div>
</div>
`;

exports[`SimpleGrid renders correctly 1`] = `
<div
  className="sc-pNWxx cmKcTU"
>
  <div
    className="sc-jrsJCI ejYQCf"
  >
    The left-hand side of the grid
  </div>
  <div
    className="sc-kEqYlL eRAHru"
  >
    The right-hand side of the grid
  </div>
</div>
`;

exports[`StatBar renders at 0% 1`] = `
<div
  className="sc-fnVYJo cmeurz"
>
  <label
    className="sc-iemXMA fmUzhZ"
  >
    LOW STAT
  </label>
  <span
    className="sc-fFSRdu daiJqc"
  >
    <span
      className="sc-bkbjAj kCyhhC"
    />
  </span>
  <label
    className="sc-iemXMA fmUzhZ"
  >
    0
  </label>
</div>
`;

exports[`StatBar renders at 50% 1`] = `
<div
  className="sc-fnVYJo cmeurz"
>
  <label
    className="sc-iemXMA fmUzhZ"
  >
    MID STAT
  </label>
  <span
    className="sc-fFSRdu daiJqc"
  >
    <span
      className="sc-bkbjAj crdRFW"
    />
  </span>
  <label
    className="sc-iemXMA fmUzhZ"
  >
    50
  </label>
</div>
`;

exports[`StatBar renders at 100% 1`] = `
<div
  className="sc-fnVYJo cmeurz"
>
  <label
    className="sc-iemXMA fmUzhZ"
  >
    HIGH STAT
  </label>
  <span
    className="sc-fFSRdu daiJqc"
  >
    <span
      className="sc-bkbjAj kCyhhC"
    />
  </span>
  <label
    className="sc-iemXMA fmUzhZ"
  >
    100
  </label>
</div>
`;

exports[`SubHeader renders correctly 1`] = `
<h2
  className="sc-gtssRu eQOHpZ"
>
  My Subheader
</h2>
`;
